### ISSUE TITLE
6809 Processor Specification incorrectly handles immediate operands

### ISSUE THREAD

**Issue by bclodfelter1024** [NationalSecurityAgency/ghidra#2116]
The pre-release 6309/6809 processor specification incorrectly processes immediate operands. The generated p-code loads the affected register with zero rather than the immediate value.

My knowledge of SLEIGH is very limited but I found by making the changes below, Load Immediate operands began to be processed by the de-compiler correctly:

Current Code:
OP1: "#"imm8    is op45=0; imm8
{
	export imm8;
}

Corrected Code:
OP1: "#"imm8    is op45=0; imm8
{
	local tmp:1 = imm8;
    export tmp;
}

Current Code:
OP2: "#"imm16    is (op47=8 | op47=0xC); imm16
{
	local imm16;
}

Corrected Code:
OP2: "#"imm16    is (op47=8 | op47=0xC); imm16
{
	local tmp:2 = imm16;
    export tmp;
}

**Comment by mumbel** [comment]
i would agree this is an issue,   `export *[const]:1 imm8` and `export *[const]:2 imm16` may yield better results.
